molecule/default/tasks/openshift_builds.yml

```diff
@@ -0,0 +1,228 @@
+- block:
+  - set_fact:
+      build_ns: "start-builds"
+      build_config: "start-build"
+      is_name: "ruby"
+      cancel_build: "cancel-build"
+
+  - name: Ensure namespace
+    kubernetes.core.k8s:
+      kind: Namespace
+      name: "{{ build_ns }}"
+
+  - name: Create ImageStream
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        apiVersion: image.openshift.io/v1
+        kind: ImageStream
+        metadata:
+          name: "{{ is_name }}"
+        spec:
+          lookupPolicy:
+            local: false
+          tags: []
+
+  - name: Create webhook secret key
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        kind: Secret
+        apiVersion: v1
+        metadata:
+          name: hooksecret
+        data:
+          WebHookSecretKey: "a4510GHAnsibleRedHat"
+
+  - name: Create build configuration
+    community.okd.k8s:
+      namespace: "{{ build_ns }}"
+      definition:
+        kind: BuildConfig
+        apiVersion: build.openshift.io/v1
+        metadata:
+          name: "{{ build_config }}"
+        spec:
+          triggers:
+            - type: "Generic"
+              generic:
+                secret: hooksecret
+          source: 
+            dockerfile: |
+              FROM openshift/ruby-22-centos7
+              USER ansible
+          strategy:
+            type: Docker
+          output: 
+            to:
+              kind: "ImageStreamTag"
+              name: "{{ is_name }}:latest"
+
+  - name: List webhook from Build configuration
+    community.okd.openshift_build_webhook_info:
+      namespace: "{{ build_ns }}"
+      build_config_name: "{{ build_config }}"
+    register: hooks
+
+  - name: Validate Generic Webhook
+    assert:
+      that:
+        - hooks.generic | length == 1
+
+  - name: Start Build using webhook until build has completed
+    community.okd.openshift_build:
+      namespace: "{{ build_ns }}"
+      build_webhook: "{{ hooks.generic.0 | regex_replace('^(.*)(<secret>)/generic$', '\\1hooksecret/generic') }}"
+      wait: yes
+      wait_timeout: 300
+    register: webhook_build
+
+  - name: Assert that new build has been created
+    assert:
+      that:
+        - webhook_build is changed
+        - webhook_build.builds.0.metadata.name == "{{ build_config }}-1"
+        - webhook_build.builds.0.status.phase == 'Complete'
+
+  - name: Start Build from Build configuration
```

Comment:
Could you also please add more tasks running in `check_mode` in general?

Reply:
yes will try to do that, got it thanks